test/integration/annotation/annotated_resource.go

```diff
@@ -0,0 +1,285 @@
+	// Deploys a static set of resources
+	t.Logf("Deploying resources")
+
+	pub, _ := testRunner.GetPublicContext(1)
+	prv, _ := testRunner.GetPrivateContext(1)
+
+	// Deploys the same set of resources against both clusters
+	// resources will have index (1 or 2), depending on the
+	// cluster they are being deployed to
+	for i, cluster := range []*client.VanClient{pub.VanClient, prv.VanClient} {
+		clusterIdx := i + 1
+
+		// Annotations (optional) to deployment and services
+		depAnnotations := map[string]string{}
+		svcNoTargetAnnotations := map[string]string{}
+		svcTargetAnnotations := map[string]string{}
+		populateAnnotations(clusterIdx, depAnnotations, svcNoTargetAnnotations, svcTargetAnnotations)
+
+		// One single deployment will be created (for the nginx http server)
+		createDeployment(t, cluster, depAnnotations)
+
+		// Now create two services. One that does not have a target address,
+		// and another that provides a target address.
+		createService(t, cluster, fmt.Sprintf("nginx-%d-svc-exp-notarget", clusterIdx), svcNoTargetAnnotations)
+		// This service with the target should not be exposed (only the target service will be)
+		createService(t, cluster, fmt.Sprintf("nginx-%d-svc-target", clusterIdx), svcTargetAnnotations)
+	}
+
+	// Wait for pods to be running
+	for _, cluster := range []*client.VanClient{pub.VanClient, prv.VanClient} {
+		t.Logf("waiting on pods to be running on %s", cluster.Namespace)
+		// Get all pod names
+		podList, err := cluster.KubeClient.CoreV1().Pods(cluster.Namespace).List(metav1.ListOptions{})
+		assert.Assert(t, err)
+		assert.Assert(t, len(podList.Items) > 0)
+
+		for _, pod := range podList.Items {
+			_, err := kube.WaitForPodStatus(cluster.Namespace, cluster.KubeClient, pod.Name, corev1.PodRunning,
+				constants.SkupperServiceReadyPeriod, constants.DefaultTick)
+			assert.Assert(t, err)
+		}
+	}
+}
+
+// populateAnnotations annotates the provide maps with static
+// annotations for the deployment, and for each of the services
+func populateAnnotations(clusterIdx int, depAnnotations map[string]string, svcNoTargetAnnotations map[string]string, svcTargetAnnotations map[string]string) {
+	// Define a static set of annotations to the deployment
+	depAnnotations[types.ProxyQualifier] = "tcp"
+	depAnnotations[types.AddressQualifier] = fmt.Sprintf("nginx-%d-dep-web", clusterIdx)
+
+	// Set annotations to the service with no target address
+	svcNoTargetAnnotations[types.ProxyQualifier] = "tcp"
+
+	// Set annotations to the service with target address
+	svcTargetAnnotations[types.ProxyQualifier] = "http"
+	svcTargetAnnotations[types.AddressQualifier] = fmt.Sprintf("nginx-%d-svc-exp-target", clusterIdx)
+}
+
+// CreateVan deploys Skupper to both clusters then generates a
+// connection token on cluster1 and connects cluster2 to cluster1
+// waiting for the network to be formed (or failing otherwise)
+func CreateVan(t *testing.T, testRunner base.ClusterTestRunner) {
```

Comment:
Just a comment:
This seems to be a duplication of:
```
func SetupSimplePublicPrivateAndConnect(ctx context.Context, r *ClusterTestRunnerBase, prefix string) error {
```

maybe you can unify them! in favor of any, the new one or the old one... I see in this implementation you are waiting for the connection to be stablished which is better IMO....

The only difference is that the other function also creates the namespaces so ... you will have to change the order to make it work (first setup skupper and then deploy the test deployments...) not sure if that causes any issue...

Reply:
I see. In this particular test I cannot do it this way, because the first test iteration expects that
all resources (deployment and services) are defined before Skupper is initialized in the namespace.

But I created #309 so that we can unify this process in a separate PR, and try to adjust the
`SetupSimplePublicPrivateAndConnect` so it can be used as a generic/common function to all
tests that use a single private and public namespaces.

Let me know what you think.

Reply:
Sounds good!